src/plugins/intel_gpu/src/plugin/ops/rnn.cpp

```diff
@@ -64,287 +67,36 @@ void GetLSTMActivationParams(const std::shared_ptr<T>& op,
-        p.add_primitive(*op, cldnn::reshape(cellInResh, inputs[2], inStateShape));
-        p.add_primitive(*op, cldnn::reorder(cellInStr, cldnn::input_info(cellInResh), hiddenLayout));
-        p.add_primitive(*op, cldnn::concatenation(input_concatID,
-                                                { permuteID, hiddenInStr },
-                                                3));
-
-        cldnn::tensor gemmSz = cldnn::tensor{ lstm_batch_size, 1, 4 * lstm_hidden_size, 1 };
-        cldnn::layout gemmLayout = cldnn::layout(lstm_dtype, cldnn::format::bfyx, gemmSz);
-        cldnn::tensor hiddenSz = cldnn::tensor{ lstm_batch_size, 1, lstm_hidden_size, 1 };
-        cldnn::tensor cellCropSz = cldnn::tensor{0, 1, 0, 0};
-
-        std::string lstm_fc_id = layerName + "_fully_connected";
-        std::string lstm_elt_id = layerName + "_lstm_elt";
-
-        cldnn::primitive_id WRconcatID = layerName + "_WRconcat";
-        p.add_primitive(*op, cldnn::concatenation(WRconcatID, { weight, recurrent }, 1));
-
-        cldnn::primitive_id FCInputReshapeID = "Reshape_bf_" + lstm_fc_id + "_for_input";
-        cldnn::tensor FCInputReshapeSz = { lstm_batch_size, inputShape.spatial[0] + inStateShape.spatial[0], 1, 1 };
-        p.add_primitive(*op, cldnn::reshape(FCInputReshapeID, cldnn::input_info(input_concatID), FCInputReshapeSz));
-
-        p.add_primitive(*op, cldnn::fully_connected(lstm_fc_id, cldnn::input_info(FCInputReshapeID), WRconcatID, bias.pid));
-        p.add_primitive(*op, cldnn::reshape(gemmReshapeID, cldnn::input_info(lstm_fc_id), gemmSz));
-        p.add_primitive(*op, cldnn::reorder(gemmReorderID, cldnn::input_info(gemmReshapeID), gemmLayout));
-        p.add_primitive(*op, cldnn::lstm_elt(lstm_elt_id, cldnn::input_info(gemmReorderID), cellInStr, clip, 0, activations,
-                                            activation_params, cldnn::lstm_weights_order::fizo, 0));
-
-
-        cldnn::tensor outSz = cldnn::tensor{ lstm_batch_size, lstm_hidden_size, 1, 1 };
-        cldnn::primitive_id outputHiddenCropID = layerName + "_hc";
-        cldnn::primitive_id outputHiddenID = layerName + ".out0";
-        p.add_primitive(*op, cldnn::crop(outputHiddenCropID, cldnn::input_info(lstm_elt_id), hiddenSz, cldnn::tensor{0, 0, 0, 0}));
-        p.add_primitive(*op, cldnn::reshape(outputHiddenID, cldnn::input_info(outputHiddenCropID), outSz), {layerName});
-
-        cldnn::primitive_id outputCellCropID = layerName + "_cc";
-        cldnn::primitive_id outputCellID = layerName + ".out1";
-        p.add_primitive(*op, cldnn::crop(outputCellCropID, cldnn::input_info(lstm_elt_id), hiddenSz, cellCropSz));
-        p.add_primitive(*op, cldnn::reshape(outputCellID, cldnn::input_info(outputCellCropID), outSz));
-    }
+    OPENVINO_ASSERT(!inputs[5].pid.empty());
+    OPENVINO_ASSERT(p.use_new_shape_infer());
+    p.add_primitive(*op, cldnn::lstm_cell(layerName+".out0", inputs[0], inputs[1], inputs[2], inputs[3], inputs[4], inputs[5], cldnn::input_info(),
```

Comment:
I think this ".out0" suffix is not needed for new shape infer

Reply:
done